hack/install-linter.sh

```diff
@@ -13,7 +13,11 @@ go version
 
 lint_version="v$(cat hack/lib/lint-version.txt)"
 
-echo "Installing golangci-lint@${lint_version}"
+# Find the toolchain version from our go.mod file. "go install" pays attention to $GOTOOLCHAIN.
+GOTOOLCHAIN=$(sed -rn 's/^toolchain (go[0-9\.]+)$/\1/p' go.mod)
+export GOTOOLCHAIN
```

Comment:
This was needed because I couldn't install the linter on my laptop without it. I was getting this output while installing the linter:

```
Installing golangci-lint@v1.63.4
go: downloading github.com/golangci/golangci-lint v1.63.4
go: github.com/golangci/golangci-lint@v1.63.4 requires go >= 1.22.1; switching to go1.22.12
go: downloading go1.22.12 (darwin/arm64)
```

And then this output while running it:
```
Error: can't load config: the Go language version (go1.22) used to build golangci-lint is lower than the targeted Go version (1.23.5)
```

The above change fixes this.